src/Compilers/CSharp/Portable/Binder/Binder_Attributes.cs

```diff
@@ -874,6 +881,19 @@ private TypedConstant GetDefaultValueArgument(ParameterSymbol parameter, Attribu
             {
                 return new TypedConstant(parameterType, kind, defaultValue);
             }
+
+            static int getArgumentIndex(ParameterSymbol parameter, ImmutableArray<string> constructorArgumentNamesOpt)
+            {
+                if (constructorArgumentNamesOpt.IsDefault || parameter.CallerArgumentExpressionParameterIndex == -1)
+                {
+                    return parameter.CallerArgumentExpressionParameterIndex;
+                }
+
+                Debug.Assert(parameter.ContainingSymbol is MethodSymbol);
+                var methodSymbol = (MethodSymbol)parameter.ContainingSymbol;
+                var callerArgumentParameter = methodSymbol.Parameters[parameter.CallerArgumentExpressionParameterIndex];
+                return constructorArgumentNamesOpt.IndexOf(callerArgumentParameter.Name);
```

Comment:
>constructorArgumentNamesOpt.IndexOf(callerArgumentParameter.Name) [](http://example.com/codeflow?start=23&length=65)

We probably should take advantage of ```GetMatchingNamedConstructorArgumentIndex``` helper rather than inventing a new way of mapping parameters to named arguments.

Reply:
@AlekseyTs `GetMatchingNamedConstructorArgumentIndex` needs to take the parameter name, which is mostly what the helper I wrote does. Also `GetMatchingNamedConstructorArgumentIndex` doesn't handle the case when no named arguments exist.

Reply:
> GetMatchingNamedConstructorArgumentIndex needs to take the parameter name, which is mostly what the helper I wrote does

I didn't mean to say the helper is all what we need, but for the same portion of the task I prefer to reuse it.

Reply:
I found that my initial approach is problematic and hard to fix it to pass the newly added test cases. So I used a totally new approach.

Reply:
> I found that my initial approach is problematic and hard to fix it to pass the newly added test cases. So I used a totally new approach.

I still think we should unify what we do here with what ```GetMatchingNamedConstructorArgumentIndex```, unless there is a good reason to diverge.

Reply:
@AlekseyTs I'm confused by this comment vs. https://github.com/dotnet/roslyn/pull/53535#discussion_r638129301

The other comment suggests to completely get rid of `GetMatchingNamedConstructorArgumentIndex`, which I did locally and all compiler tests passed.